components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -53,7 +55,12 @@ func (s *OIDCService) CreateClientConfig(ctx context.Context, req *connect.Reque
 		return nil, err
 	}
 
-	err = assertIssuerIsReachable(req.Msg.GetConfig().GetOidcConfig().GetIssuer())
+	oidcConfig := req.Msg.GetConfig().GetOidcConfig()
+	err = assertIssuerIsReachable(ctx, oidcConfig.GetIssuer())
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+	err = assertIssuerProvidesDiscovery(ctx, oidcConfig.GetIssuer())
```

Comment:
Given we're performing both of these, unconditionally, what is the value in doing both? Ensuring discovery, also checks it's reachable so we're not gaining much by separating them